### PR TITLE
feat: Spike out dummy PromptVersion connection

### LINKED ISSUE #5754: [hub] [gql] prompt versions list
For a given prompt, we want to be able to pull prompt versions. E.g.

This ticket is to wire up a node based API for the version history from the database.

```graphql
query {
   prompt: node(id: asdfasdf) {
      name
      versions {
        edges {
            template_version: node {
                  user
                  description
                  template
                  tools
            }
        }
      }
   }
}
```

```graphql
query node {
    templateVersion: node($nodeId) {
          id
         ... on PromptVersion {
              user
              description
              template
              tools
         }
    }
}

### PR DESCRIPTION
**Pull request by anticorrelator**
resolves #5754

### CODE REVIEW

File: src/phoenix/server/api/types/PromptVersion.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import strawberry
+from strawberry.relay import Node, NodeID
+from strawberry.scalars import JSON
+
+
+@strawberry.enum
+class PromptTemplateType(str, Enum):
+    TEXT = "text"
+    JSON = "json"
```
Comment (RogerHYang) [inline src/phoenix/server/api/types/PromptVersion.py:14]: is this supposed to match what's in the DB?
```
        sa.Column(
            "template_type",
            sa.String,
            sa.CheckConstraint(
                "template_type IN ('chat', 'str')",
                name="template_type",
            ),
            nullable=False,
        ),
```

File: src/phoenix/server/api/types/PromptVersion.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+@strawberry.enum
+class PromptTemplateFormat(str, Enum):
+    MUSTACHE = "mustache"
+    FSTRING = "fstring"
+
```
Comment (RogerHYang) [inline src/phoenix/server/api/types/PromptVersion.py:21]: is this missing `NONE`?

**Review by RogerHYang — APPROVED** [review]

File: src/phoenix/server/api/types/PromptVersion.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+@strawberry.enum
+class PromptTemplateType(str, Enum):
+    TEXT = "text"
+    JSON = "json"
+
+
+@strawberry.enum
+class PromptTemplateFormat(str, Enum):
+    MUSTACHE = "mustache"
+    FSTRING = "fstring"
```
Comment (mikeldking) [inline src/phoenix/server/api/types/PromptVersion.py:20]:
```suggestion
    FSTRING = "fstring"
    NONE = "none"
```

File: src/phoenix/server/api/types/PromptVersion.py (hunk 1 above)
Comment (mikeldking) [reply src/phoenix/server/api/types/PromptVersion.py:14]: added

File: app/schema.graphql [hunk 4]
```diff
@@ -1402,6 +1402,49 @@ type PromptResponse {
...
+}
+
+type PromptVersion implements Node {
+  """The Globally Unique ID of this object"""
+  id: GlobalID!
+  user: String
+  description: String!
+  templateType: PromptTemplateType!
+  templateFormat: PromptTemplateFormat!
+  template: JSON!
```
Comment (mikeldking) [inline app/schema.graphql:1422]: I think we can make this a union type and give it a bit more structure

File: src/phoenix/server/api/queries.py [hunk 5]
```diff
@@ -612,6 +617,85 @@ def prompts(
             args=args,
         )
 
+    @strawberry.field
+    async def prompt_versions(
```
Comment (mikeldking) [inline src/phoenix/server/api/queries.py:621]: this goes on prompt

**Pull request merged**